google/cloud/bigquery/magics/magics.py

```diff
@@ -720,6 +736,39 @@ def _cell_magic(line, query):
     finally:
         close_transports()
 
+    if args.send_widget_job:
+        client = bigquery.Client(
+            project=project,
+            credentials=context.credentials,
+            default_query_job_config=context.default_query_job_config,
+            client_info=client_info.ClientInfo(user_agent=IPYTHON_USER_AGENT),
+            client_options=bigquery_client_options,
+        )
+        job_config = bigquery.job.query.QueryJobConfig()
+        job_config.query_parameters = params
+
+        widget_job = client.query(query, job_config=job_config)
+
+        def thread_func(widget_job, out):
+            time_sec = 10.0
+            while widget_job.state != "DONE":
+                job_status = "Job is still running!"
+                out.append_stdout(f"{job_status}")
+                time.sleep(time_sec)
+                widget_job.reload()
+            else:
+                result = widget_job.to_dataframe()
+            out.append_stdout(f"{result}")
```

Comment:
I'm unfamiliar with the widget abstraction, but this seems odd to me.  the result dataframe is simply printed?  Don't users want to access the result programmatically?

Reply:
I'm running and tweaking in Jupyterlab! Currently it's supposed to display the widget while the job is running and then at the end print the result to dataframe.